documentation/query-api.md

```diff
@@ -0,0 +1,213 @@
+## Status Query API, v1
+
+You can query for status information about services running in your application
+by making an HTTP request to the `/status` endpoint.
+
+## Status Detail Level
+
+When querying for service status, you may optionally request a specific level
+of detail to be returned.  The valid levels are:
+
+* `"critical"` : returns only the bare minimum amount of status information for
+  each service.  Intended to return very quickly and to be suitable for use
+  cases like health checks for a load balancer.
+* `"info"` : typically returns a bit more info than the `"critical"` level would,
+  for each service.  The specific data that is returned will depend on the
+  implementation details of the services in your application, but should generally
+  be data that is useful for a human to get a quick impression of the health / status
+  of each service.
+* `"debug"` : this level can be used to request very detailed status information
+  about a service, typically used by a human for debugging.  Requesting this
+  level of status information may be significantly more expensive than the lower
+  levels, depending on the service.  A common use case would be for a service to
+  provide some detailed aggregate metrics about the performance or resource
+  usage of its subsystems.
+
+The information returned for any service at each increasing level of detail should
+ be additive; in other words, `"info"` should return the same data structure as `"critical"`,
+ but may add additional data in the `status` field.  Likewise, `"debug"` should
+ return the same data structure as `"info"`, but may add additional information
+ in the `status` field.
+
+## `GET /status/v1/services`
+
+This will return status information for all registered services in an application.
+
+### URL Parameters
+
+* `level`: Optional.  A JSON String from among the legal
+[Status Detail Levels](#StatusDetailLevel) listed above.  Status information for
+all registered services will be provided at the requested level of detail.  If
+not provided, the default level is `"info"`.
+
+### Response Format
+
+The response format will be a JSON _Object_, which will look something like this:
+
+    {<service-name>: {
+        "service_version": <service-version>,
+        "service_status_version": <service-status-version>,
+        "detail_level": <detail-level>,
+        "is_running": <service-state>,
+        "status": <any>
+        },
+     <service-name>: {
+        ...
+        },
+    ...
+    }
+
+For detailed information, please see the [Wire Format Specification](./wire-formats.md).
+
+NOTE: If any services in your application have registered more than one supported
+status format version, and you are querying at a detail level other than `critical`
+this endpoint will *always* return the latest format.  Therefore,
```

Comment:
Yeah, sorry, that's poor phrasing on my part.  What I've been thinking is that if you specify `critical` as the level, then the `status` field would always be `null`.  (We don't enforce that, and we should probably chat about it to decide whether it's even a good idea, but that was the assumption that I had in my head when writing these docs.)  If that were the case, then theoretically there's no schema left that could vary between format versions, so you wouldn't really need to think about which format version you were requesting / viewing.

This is phrased poorly, regardless, and it probably makes sense to just remove the stanza about 'other than `critical`' for now.